File: flutter/lib/models/translation_result.dart
/// A translation result for a transcript segment.
class TranslationResult {
  final int segmentId;
  final String translatedText;
  final String targetLang;

  TranslationResult({
    required this.segmentId,
    required this.translatedText,
    required this.targetLang,
  });
}

File: kmp/shared/src/commonMain/kotlin/com/sondt/justtranscribe/TranslationResult.kt
package com.sondt.justtranscribe

import kotlinx.serialization.Serializable

/** A translation of [segmentId]'s text into [targetLang]. */
@Serializable
data class TranslationResult(
    val segmentId: Int,
    val translatedText: String,
    val targetLang: String,
)
